question-service/app/crud/questions.py

```diff
@@ -24,67 +38,54 @@ async def create_question(question: CreateQuestionModel) -> QuestionModel:
     new_question = await question_collection.insert_one(question.model_dump())
     return await question_collection.find_one({"_id": new_question.inserted_id})
 
-async def get_all_questions(category: str, complexity: str, search: str) -> QuestionCollection:
+async def get_all_questions(
+    categories: List[CategoryEnum],
+    complexity: ComplexityEnum,
+    search: str
+) -> QuestionCollection:
     query = {}
     if search:
         query["title"] = {"$regex": search, "$options": "i"}
-    if category:
-        query["category"] = {"$regex": category, "$options": "i"}
+    if categories:
+        query["categories"] = {"$all": categories}
     if complexity:
         query["complexity"] = complexity
     questions = await question_collection.find(query).to_list(1000)
-
     return QuestionCollection(questions=questions)
 
 async def get_question_by_id(question_id: str) -> QuestionModel:
-    try:
-        object_id = ObjectId(question_id)
-    except InvalidId:
-        raise InvalidQuestionIdError(question_id)
-    
+    object_id: ObjectId = _parse_oid(question_id)
     existing_question = await question_collection.find_one({"_id": object_id})
     if existing_question is None:
         raise QuestionNotFoundError(question_id)
-    return existing_question
+    return QuestionModel.parse_obj(existing_question)
 
 async def delete_question(question_id: str) -> MessageModel:
-    try:
-        object_id = ObjectId(question_id)
-    except InvalidId:
-        raise InvalidQuestionIdError(question_id)
-    
-    existing_question = await question_collection.find_one({"_id": object_id})
```

Comment:
I think we should still include the checking of the existence of the question before deleting. Or else, the `QuestionNotFound` exception in `routers/questions.py` will never be thrown and caught also.

Reply:
The exceptions are still thrown and caught.

`InvalidQuestionIdError`
![image](https://github.com/user-attachments/assets/e0d8acaa-68d0-4764-b540-41aa1a24fc60)

`QuestionNotFoundError`
![image](https://github.com/user-attachments/assets/8fc035bb-be12-4775-8f7e-db9c4d9d19f9)